modules/frontend/traceid_handlers.go

```diff
@@ -60,7 +49,16 @@ func newTraceIDHandler(cfg Config, next pipeline.AsyncRoundTripper[combiner.Pipe
 			"tenant", tenant,
 			"path", req.URL.Path)
 
-		comb := combinerFn(o.MaxBytesPerTrace(tenant), marshallingFormat)
+		var traceRedactor combiner.TraceRedactor
+		if dataAccessController != nil {
+			traceRedactor, err = dataAccessController.TraceByIDResponseRedactor(req)
+			if err != nil {
+				level.Error(logger).Log("msg", "trace id: failed to get trace redactor", "err", err)
+				return httpInvalidRequest(err), nil
```

Comment:
wouldn't this be a 500 instead of a 400?

Reply:
this is tricky because for our implementation, the only errors that get tripped are due to bad request- but if we were to make it more generic a 500 is more fitting. what do you think?